### DISCUSSION TITLE
Team Sharing - Permission Management

### DISCUSSION THREAD

**Discussion by andy1463** [1Remote/1Remote#386] [Ideas]
Hello,
would be great if there will be a possibility to hide different connections for different members in team sharing because not every team member should have access to every server.

Thanks

**Comment by majkinetor** [comment]
This is IMO out of scope of this project. Identity and access management features (IAM) would reaquire a lot of aditional development and careful design. There are all sorts of things to do for that instead of just hiding some items. The project was not ment to be corporative tool as signified by its original name. Also, project currently has single developer @VShawn which needs to have a life.

On the other hand, it would be good to have it, but I personaly wouldn't go that way until lets say v2+. There is still lot of work to be done in the main domain. Nothing stops specific team to use their own database (which could be on the same central database server). One could even create sharing procedure between "main database" that contains all connections, and team databases which contain cherry picked copies of the main database connections. That can be done by database admin alone with some stored procedures which could be sugar coated with some CLI tools or shell scripts for better UX.  

Real thing needs to have options to define teams, sharing to teams, connection and secrets sharing, tag sharing, revoking, timed access, editing/viewing permissions, etc. My brain hurts while trying to think what exactly goes into a good IAM features. Not exactly trivial.

**Comment by VShawn** [comment]
Hi Andy1463,

Thanks for your suggestion on improving the project. 

With the team sharing feature in 1Remote, you can easily share access to specific servers with your team members. However it offers only very basic access control, and is not designed to provide individualized access control for each team member (at least for now).

In our current implementation, team members can be added to specific groups (based on mysql users), and each group can be given access to specific resources. This makes 1Remote provide granular permissions settings, which allow you to define who can view, edit, or delete servers.

See https://1remote.github.io/usage/team/team-sharing/ for more detail.

As @majkinetor :kissing_heart: has pointed out, IAM features would require a lot of additional development and careful design. While it's true that this feature would be useful, we still need to focus on other sides.
